cmdeploy/src/cmdeploy/dovecot/remove-seen.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+"""Remove seen messages that are older than two days
+if maildir has more than 80 MB of messages."""
+import sys
+import time
+from pathlib import Path
+
+
+def getdirsize(path):
+    return sum(f.stat().st_size for f in path.glob("**/*") if f.is_file())
+
+
+def parse_dovecot_seen(path):
+    return "S" in path.name.split(":2,")[-1]
+
+
+def main():
+    now = time.time()
+
+    mailhome = Path(sys.argv[1])
+
+    for p in mailhome.iterdir():
+        dirsize = getdirsize(p / "cur") + getdirsize(p / "new")
+        if dirsize < 80000000:
+            continue
+
+        removed_bytes = 0
+        for mailpath in (p / "cur").iterdir():
+            seen = parse_dovecot_seen(mailpath)
+            stat = mailpath.stat()
+            size = stat.st_size
+            if seen and now > stat.st_mtime + 2 * 24 * 3600:
+                removed_bytes += size
+                mailpath.unlink(missing_ok=True)
+
+        if removed_bytes > 0:
+            (p / "maildirsize").unlink(missing_ok=True)
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
when #344 is in, this should be rewritten using "doveadm --f flow quota get -A" to get all quotas from users 
and then `doveadm expunge  -A mailbox INBOX SAVEDBEFORE 1d \( SEEN \)` for each user that is above 80%

Reply:
here is a rough script that does it: 
```
import subprocess


def get_near_quota_users():
    res = subprocess.check_output(["doveadm", "-f", "flow", "quota", "get", "-A"]).decode("ascii")
    for line in res.split("\n"):
        parts = line.split()
        if not parts:
            continue
        if parts[4].lower() != "type=storage":
            continue
        addr = parts[0]
        c, percent = parts[-1].split("=")
        assert c == r"%"
        if int(percent) > 80:
            yield addr 

def expunge_seen_messages_from_near_quota_users():
    for addr in get_near_quota_users():
        print("trying to expunge seen messages", addr)
        subprocess.run(["doveadm", "expunge", "-u", addr, "mailbox", "INBOX", "(", "SEEN", ")"], check=True)
        subprocess.run(["doveadm", "purge", "-u", addr])

expunge_seen_messages_from_near_quota_users()
```